fxa/requests.py

```diff
@@ -0,0 +1,71 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+from __future__ import absolute_import
+
+from binascii import hexlify
+from hashlib import sha256
+import os
+
+from requests.auth import AuthBase
+from six.moves.urllib.parse import urlparse
+
+from fxa.core import DEFAULT_SERVER_URL, Client
+
+
+class FxABrowserIdAuth(AuthBase):
+    """Handles authentication using FxA BrowserID.
+
+    :param email:
+      The user Firefox Account email address
+
+    :param password:
+      The user Firefox Account password
+
+    :param audience:
+      The audience the assertion is valid for.
+
+    :param server_url:
+      The url of the Firefox Account server.
+
+    """
+    def __init__(self, email, password, audience=None,
+                 server_url=DEFAULT_SERVER_URL):
+        self.email = email
+        self.password = password
+        self.audience = audience
+        self.server_url = server_url
+
+    def __call__(self, request):
+        client = Client(server_url=self.server_url)
+        session = client.login(self.email, self.password, keys=True)
+
+        if self.audience is None:
+            url = urlparse(request.url)
+            self.audience = "%s://%s/" % (url.scheme, url.netloc)
```

Comment:
What's the use-case for providing a custom 'audience' here?  In general it's not great to be sending an assertion with audience "A" to a site on domain "B", so I'm wondering if there's a different way to achieve things here.